addons/knowledge/models/knowledge_article.py

```diff
@@ -612,7 +614,7 @@ def get_user_sorted_articles(self, search_query, fields, order_by, limit):
             - Favourite count
         and returned result mimic a search_read result structure.
         """
-        search_domain = ["|", ("name", "ilike", search_query), ("parent_id.name", "ilike", search_query)]
+        search_domain = ["|", ("name", "ilike", search_query), ("main_article_id.name", "ilike", search_query)]
         articles = self.search(search_domain, order=order_by, limit=limit)
```

Comment:
if we don't have access to main_article_id.display_name (as we need to sudo it later), this search would also need a sudo, right ?

Reply:
Weirdly enough, no. For some reason works here without sudo.